metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/utils/SyslogUtils.java

```diff
@@ -0,0 +1,89 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.metron.parsers.utils;
+
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+
+public class SyslogUtils {
+
+    public static long convertToEpochMillis(String logTimestamp, String logTimeFormat) {
+        ZonedDateTime timestamp = ZonedDateTime.parse(logTimestamp, DateTimeFormatter.ofPattern(logTimeFormat).withZone(ZoneOffset.UTC));
+        return timestamp.toEpochSecond() * 1000;
+    }
+
+    public static long parseTimestampToEpochMillis(String logTimestamp) {
+        if (logTimestamp.length() < 20) {
+            ZonedDateTime now = ZonedDateTime.now(ZoneOffset.UTC);
+            int year = now.getYear();
+            if (now.getDayOfYear() == 1 && !now.getMonth().toString().substring(0,3).equals(logTimestamp.substring(0,3).toUpperCase()))
+                year--;
```

Comment:
Gotcha.  So anything that comes in on the first day of the year, with a month that is not January, will be backdated.

If something comes in on the 2nd day of the year, with a month of December, it will NOT be backdated.  The period of time that we are willing to backdate, is effectively 1 day currently.  

Maybe that time period needs to be configurable.  The user defines the period of time, 1 day, 2 days, 1 week, after the beginning of the year in which messages can possibly be backdated.

Are there certain conditions under which the logic should blow-up and error?  What if we are going to backdate a message where the month is July?  Should we just do that or should we error?